extensions/github/src/auth.ts

```diff
@@ -54,3 +54,11 @@ export function getOctokit(): Promise<Octokit> {
 	return _octokit;
 }
 
+type CreateRepositoryResponseData = Awaited<ReturnType<Octokit['repos']['createForAuthenticatedUser']>>['data'];
+
+export function getRemoteUrl(githubRepository: CreateRepositoryResponseData) {
+	const config = workspace.getConfiguration('git');
+	const remoteUrlType = config.get<'https' | 'ssh'>('remoteUrlType');
+	const { clone_url, ssh_url } = githubRepository;
+	return remoteUrlType === 'https' ? clone_url : ssh_url;
+}
```

Comment:
Instead of creating this "util" which just introduces spaghetti, simply read the config in the places you need to make such a decision and pick the right URL right in there. It's fine if there's duplicated code.